apps/dashboard/src/pages/integrations/components/hooks/use-integration-primary-modal.tsx

```diff
@@ -0,0 +1,105 @@
+import { useState } from 'react';
+import { CHANNELS_WITH_PRIMARY, IIntegration, ChannelTypeEnum } from '@novu/shared';
+import { IntegrationFormData } from '../../types';
+import { handleIntegrationError } from '../utils/handle-integration-error';
+
+type UseIntegrationPrimaryModalProps = {
+  onSubmit: (data: IntegrationFormData, skipPrimaryCheck?: boolean) => Promise<void>;
+  integrations?: IIntegration[];
+  integration?: IIntegration;
+  channel?: ChannelTypeEnum;
+  mode: 'create' | 'update';
+};
+
+export function useIntegrationPrimaryModal({
+  onSubmit,
+  integrations = [],
+  integration,
+  channel,
+  mode,
+}: UseIntegrationPrimaryModalProps) {
+  const [isPrimaryModalOpen, setIsPrimaryModalOpen] = useState(false);
+  const [pendingData, setPendingData] = useState<IntegrationFormData | null>(null);
+  const isChannelSupportPrimary = CHANNELS_WITH_PRIMARY.includes(
+    integration?.channel ?? channel ?? ChannelTypeEnum.EMAIL
+  );
+  const shouldShowPrimaryModal = (data: IntegrationFormData) => {
+    if (!channel && !integration) return false;
+
+    const hasSameChannelActiveIntegration = integrations?.some(
+      (el) =>
+        (mode === 'update' ? el._id !== integration?._id : true) &&
+        el.active &&
+        el.channel === (integration?.channel ?? channel)
+    );
```

Comment:
The primary integration should be at the same time active. I'm not sure if we need to check for same channel active integration. Can you please clarify?